pkg/controller/volume/attacherdetacher/attacher_detacher.go

```diff
@@ -119,15 +119,23 @@ func (ad *attacherDetacher) generateAttachVolumeFunc(
 
 		if attachErr != nil {
 			// On failure, just log and exit. The controller will retry
-			glog.Errorf("Attach operation for %q failed with: %v", volumeToAttach.VolumeName, attachErr)
+			glog.Errorf(
+				"Attach operation for device %q to node %q failed with: %v",
+				volumeToAttach.VolumeName, volumeToAttach.NodeName, attachErr)
 			return attachErr
 		}
 
+		glog.Infof(
+			"Successfully attached device %q to node %q. Will update actual state of world.",
+			volumeToAttach.VolumeName, volumeToAttach.NodeName)
+
 		// Update actual state of world
 		_, addVolumeNodeErr := actualStateOfWorld.AddVolumeNode(volumeToAttach.VolumeSpec, volumeToAttach.NodeName)
 		if addVolumeNodeErr != nil {
 			// On failure, just log and exit. The controller will retry
-			glog.Errorf("Attach operation for %q succeeded but updating actualStateOfWorld failed with: %v", volumeToAttach.VolumeName, addVolumeNodeErr)
+			glog.Errorf(
```

Comment:
Node will be happy since it will see the disk show up and will assume it is attached, which it is. The controller will assume that it is still not attached, and will continue trying to attach it (retrying), which will succeed since the disk is already attached and hopefully it won't fail here on the retry, if it does, it'll retry.